### PR TITLE
fix: Configure the PortalNavigationIconUpgradePlugin to set icons to the portal navigation nodes - EXO-69848

### PR DESCRIPTION
**Pull request by sofyenne**
Prior to this change after migrating from the 6.4 to the 6.5 then to the 6.6 version , the external stream node navigation icon was incorrect , this change is going to configure the PortalNavigationIconUpgradePlugin in order to set icons for the portal navigation nodes .

### CODE REVIEW

File: data-upgrade-navigations/src/main/resources/conf/portal/configuration.xml [hunk 1]
```diff
@@ -195,5 +195,48 @@
...
+      <description>Configure portal node icons</description>
+      <init-params>
+        <value-param>
+          <name>product.group.id</name>
+          <description>The groupId of the product</description>
+          <value>org.exoplatform.platform</value>
+        </value-param>
+        <value-param>
+          <name>plugin.upgrade.target.version</name>
+          <description>The plugin target version (will not be executed if previous version is equal or higher than 6.5.0)</description>
```
Comment (azayati) [inline data-upgrade-navigations/src/main/resources/conf/portal/configuration.xml]:
```suggestion
          <description>The plugin target version (will not be executed if previous version is equal or higher than 6.6.0)</description>
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: data-upgrade-navigations/src/main/resources/conf/portal/configuration.xml (hunk 1 above)
Comment (sofyenne) [reply data-upgrade-navigations/src/main/resources/conf/portal/configuration.xml]: done !

**Review by azayati — CHANGES_REQUESTED** [review]
Can you try to use this UP instead https://github.com/exoplatform/digital-workplace/blob/develop/digital-workplace-webapps/src/main/webapp/WEB-INF/conf/digital-workplace/portal-upgrade-configuration.xml#L90 with 
```
<field name="updateNavigation">
  <boolean>true</boolean>
</field>
```

File: data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    NodeEntity nodeEntity = new NodeEntity();
+    nodeEntity.setName("external-stream");
+    nodeEntity.setIcon(null);
+    nodeEntity.setPage(pageEntity);
+    nodeEntity.setTarget(NodeTarget.NEW_TAB);
+    nodeDAO.create(nodeEntity);
+    nodeEntity = nodeDAO.findAllByPage(pageEntity.getId()).stream().filter(e -> e.getName().equals("external-stream")).toList().get(0);
+    assertNotNull(nodeEntity);
+    //
+    assertNotNull(nodeEntity);
```
Comment (azayati) [inline data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java]:
```suggestion
    assertNotNull(nodeEntity);
```

File: data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  private NodeDAO                       nodeDAO;
+
+  @Before
+  public void setUp() {
+    container = getContainer();
+    siteDAO = container.getComponentInstanceOfType(SiteDAO.class);
+    pageDAO = container.getComponentInstanceOfType(PageDAO.class);
+    nodeDAO = container.getComponentInstanceOfType(NodeDAO.class);
+    entityManagerService = container.getComponentInstanceOfType(EntityManagerService.class);
+    RequestLifeCycle.begin(container);
```
Comment (azayati) [inline data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java]:
```suggestion
    begin();
```

File: data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    portalNodeIconsValueParam.setValue("fas fa-user-lock");
+    initParams.addParameter(productGroupIdValueParam);
+    initParams.addParameter(portalNodeNamesValueParam);
+    initParams.addParameter(portalNodeIconsValueParam);
+    this.portalNavigationIconMigration = new PortalNavigationIconMigration(entityManagerService, initParams);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    RequestLifeCycle.end();
```
Comment (azayati) [inline data-upgrade-navigations/src/test/java/org/exoplatform/migration/PortalNavigationIconMigrationTest.java]:
```suggestion
    end();
```

**Review by azayati — CHANGES_REQUESTED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**